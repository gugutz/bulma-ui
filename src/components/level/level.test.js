import React from 'react'
import renderer from 'react-test-renderer'
import Level from './level'

describe('<Level /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Level>
          <Level.Left>
            <Level.Item>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Left>
          <Level.Right>
            <Level.Item>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Right>
        </Level>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Level should support isMobile ', () => {
    const tree = renderer
      .create(
        <Level isMobile>
          <Level.Left>
            <Level.Item>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Left>
          <Level.Right>
            <Level.Item>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Right>
        </Level>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Item should support hasTextCentered prop', () => {
    const tree = renderer
      .create(
        <Level isMobile>
          <Level.Left>
            <Level.Item hasTextCentered>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Left>
          <Level.Right>
            <Level.Item hasTextCentered>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Right>
        </Level>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Item should support isNarrow prop', () => {
    const tree = renderer
      .create(
        <Level isMobile>
          <Level.Left>
            <Level.Item isNarrow>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Left>
          <Level.Right>
            <Level.Item isNarrow>item</Level.Item>
            <Level.Item>item</Level.Item>
          </Level.Right>
        </Level>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
