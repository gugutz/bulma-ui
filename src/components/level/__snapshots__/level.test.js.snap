// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Level /> component Item should support hasTextCentered prop 1`] = `
<div
  className="level is-mobile"
>
  <div
    className="level-left"
  >
    <div
      className="level-item has-text-centered"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item has-text-centered"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
</div>
`;

exports[`<Level /> component Item should support isNarrow prop 1`] = `
<div
  className="level is-mobile"
>
  <div
    className="level-left"
  >
    <div
      className="level-item is-narrow"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item is-narrow"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
</div>
`;

exports[`<Level /> component Level should support isMobile  1`] = `
<div
  className="level is-mobile"
>
  <div
    className="level-left"
  >
    <div
      className="level-item"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
</div>
`;

exports[`<Level /> component should render correctly 1`] = `
<div
  className="level"
>
  <div
    className="level-left"
  >
    <div
      className="level-item"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item"
    >
      item
    </div>
    <div
      className="level-item"
    >
      item
    </div>
  </div>
</div>
`;
